### PR TITLE
[Model Monitoring] Add data drift application

### PR DESCRIPTION
**Pull request by jond01**
Resolves [ML-5766](https://jira.iguazeng.com/browse/ML-5766).

This PR adds the basic version of the data drift application ported from the [model monitoring batch](https://github.com/mlrun/mlrun/blob/v1.7.0-rc1/mlrun/model_monitoring/batch.py) job.
The app returns four metrics for data drift based on the reference versus actual histograms:
* TVD,
* Hellinger,
* KL divergence (no status),
* The average of TVD & Hellinger as the general drift result.

### CODE REVIEW

**Review by Eyal-Danieli — COMMENTED** [review]
Great job!
Just update the internal imports (use `import` instead of `from... import ...`).
Important to note that I didn't test it in E2E flow but I agree with @jond01 that it should be tested after we implement the drift app in the general flow - https://jira.iguazeng.com/browse/ML-5759

File: mlrun/model_monitoring/applications/data_drift.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+)
+from mlrun.model_monitoring.application import (
+    ModelMonitoringApplicationBase,
+    ModelMonitoringApplicationResult,
+)
+from mlrun.model_monitoring.batch import (
+    HellingerDistance,
+    HistogramDistanceMetric,
+    KullbackLeiblerDivergence,
+    TotalVarianceDistance,
```
[the remark this answers is not in the document]
Comment (Eyal-Danieli) [reply mlrun/model_monitoring/applications/data_drift.py:33]: Just saw https://github.com/mlrun/mlrun/pull/5182 :)

File: mlrun/model_monitoring/applications/data_drift.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Final, Optional, Protocol
+
+import numpy as np
+from pandas import DataFrame, Timestamp
+
+from mlrun.common.schemas.model_monitoring.constants import (
```
[the remark this answers is not in the document]
Comment (jond01) [reply mlrun/model_monitoring/applications/data_drift.py:21]: After a second thought, I think that the import from `mlrun.common` is safe and can be regarded as an "external" subpackage thanks to this rule:
https://github.com/mlrun/mlrun/blob/ca0de3c504c591207116aa6ce294026695ea03bd/pyproject.toml#L42-L47

As for the other imports here - they are also safe IMO - no way for a circular import, as the application should be treated as "user code" and should not be imported by other parts in the SDK.
I suggest we further discuss this rule in the broader forum before adhering to it.
What do you think?

[the remark this answers is not in the document]
Comment (Eyal-Danieli) [reply mlrun/model_monitoring/applications/data_drift.py:21]: Alright, though even if these packages are "safe" we should keep consistency across MLRun code. Anyway, as you suggested, let's discuss it in the next committee meeting.

**Review by Eyal-Danieli — APPROVED** [review]

**Pull request merged**